Inverted Index Application Build

abdulsemiu-atanda opened the pull request:
**What does this PR do?**
Implements Elastic search inverted index algorithm

**How can this PR be manually tested?**
This PR can be tested [here](aatanda-inverted-index.herokuapp.com)

andela-abdulsemiu committed:
feature(setup): inverted-index.js file in src folder fix

andela-abdulsemiu committed:
feature(migration): code migration from es5 to es6

andela-abdulsemiu committed:
separation of jquery code into a separate file

andela-abdulsemiu committed:
jshint configuration file for HoundCI fix

andela-abdulsemiu committed:
cleanup of jquery code

andela-abdulsemiu committed:
more jshint configuration fix

andela-abdulsemiu committed:
view update

andela-abdulsemiu committed:
source files fix

andela-abdulsemiu committed:
new file inverted-index-spec.js fix

andela-abdulsemiu committed:
jquery file for user experience fix

andela-abdulsemiu committed:
cleanup for json file keywords occurence fix

andela-abdulsemiu committed:
search button on view fix

andela-abdulsemiu committed:
Update for jshint config file with angular

andela-abdulsemiu committed:
Angular directives in view fix

andela-abdulsemiu committed:
gulp,browser-sync fix for auto reload of changed files

andela-abdulsemiu committed:
Framework switch from jquery to angularjs

andela-abdulsemiu committed:
Gulp taskrunner fix

andela-abdulsemiu committed:
Removal of duplication of functionality
unnecessary function removal

andela-abdulsemiu committed:
Page Controller fix after migration to angularjs

andela-abdulsemiu committed:
HTML elements css for index.html

andela-abdulsemiu committed:
Additional sample JSON file

andela-abdulsemiu committed:
TravisCI configuration file

andela-abdulsemiu committed:
Karma spec runner configuration file

andela-abdulsemiu committed:
Additional dependencies fix

andela-abdulsemiu committed:
Code Refactor fix

andela-abdulsemiu committed:
Spec update

andela-abdulsemiu committed:
Super User access fix

andela-abdulsemiu committed:
Travis configuration update

andela-abdulsemiu committed:
Readme file update

andela-abdulsemiu committed:
Travis Build badge fix

andela-abdulsemiu committed:
Move gulp from devDependencies to dependencies

andela-abdulsemiu committed:
Heroku deployment error fix

andela-abdulsemiu committed:
port fix for heroku

andela-abdulsemiu committed:
gulpfile update

andela-abdulsemiu committed:
gulpfile update

andela-abdulsemiu committed:
dependency update

andela-abdulsemiu committed:
server.js for Heroku

andela-abdulsemiu committed:
unnecessary line of code delete

andela-abdulsemiu committed:
Readme file update

andela-abdulsemiu committed:
package.json file update

abdulsemiu-atanda committed:
Update README.md

abdulsemiu-atanda committed:
Update README.md

abdulsemiu-atanda committed:
Update README.md

andela-abdulsemiu committed:
Duplication removal fix

andela-abdulsemiu committed:
Unnecessary Redundancy fix

andela-abdulsemiu committed:
Code Climate configuration file

andela-abdulsemiu committed:
Merge branch 'development' of https://github.com/andela-aatanda/checkpoint1-inverted-index into development

andela-abdulsemiu committed:
Codeclimate test reporter dependency

andela-abdulsemiu committed:
codeclimate test reporter after script fix

andela-abdulsemiu committed:
Karma-coverage dependency update

andela-abdulsemiu committed:
Coverage report fix

andela-abdulsemiu committed:
Coverage report directory update

andela-abdulsemiu committed:
Coverage badge update

andela-abdulsemiu committed:
revert to last successful travis build

andela-abdulsemiu committed:
test coverage after script fix

andela-abdulsemiu committed:
Indentation fix

andela-abdulsemiu committed:
HoundCI configuration file

andela-abdulsemiu committed:
Configuration file update

andela-abdulsemiu committed:
Indentation Fix

thuo commented on index.html:
```diff
@@ -0,0 +1,85 @@
+        <link rel="stylesheet" href="css/index.css">
+        <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
+        <link rel="stylesheet" href="https://code.getmdl.io/1.2.1/material.indigo-pink.min.css">
+        <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Roboto:300,400,500,700" type="text/css">
```
Change the protocol to https so that the request is not blocked when the entire page is loaded over https e.g. on heroku.

You can actually remove the protocol part in all of the and it will default to the protocol used by the page e.g. `<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Roboto:300,400,500,700" type="text/css">`

thuo commented on jasmine/spec/inverted-index-spec.js:
```diff
@@ -0,0 +1,96 @@
+beforeEach(function() {
+    this.indexInstance = new Index();
+  });
+
```
Nitpick: `beforeEach` has different indentation level from the `it` below. To imporve code quality, strive to have consistent indentation to make the code readable.

thuo commented on src/inverted-index.js:
```diff
@@ -0,0 +1,77 @@
+            let merge = [];
+            let keywords =[];
+            let mySet;
+            for (let i=0; i <file.length; i++){
```
_Nitpick_: Consistency in stuff such the space between the operators should be consistent. e.g. if you put a space before `<` you should also put a space after.

While this may seem trivial it's a good habit to form as most real-world projects will require you to stick to such rules.

andela-abdulsemiu committed:
Indentation Fix

azemoh commented on src/pageCtrl.js:
```diff
@@ -0,0 +1,53 @@
+    $scope.fileUpload = function() {
+        let fileDetails = document.getElementById('file').files[0];
+        if(fileDetails===undefined){
+            console.log('No file chosen');
```
Good job handling edge cases. It would be nicer if you notify the user about failures instead of just logging them to the console.

azemoh commented on src/inverted-index.js:
```diff
@@ -0,0 +1,77 @@
+                let clean = tokenize(file[i].text);
+                merge.push(clean.split(' '));
+                keywords = [].concat.apply([], merge);
+                mySet = new Set(keywords);
```
Good use of `Set` to get unique tokens.

azemoh commented on src/pageCtrl.js:
```diff
@@ -0,0 +1,53 @@
+    };
+    $scope.search = function() {
+        let terms = document.getElementById('terms').value;
+        $scope.docArray = read.searchIndex(terms,$scope.docArray);
```
**Personal style preference:** I think spacing between function arguments and empty lines after code blocks increases code readability.

azemoh reviewed (commented):
Good job @andela-aatanda. LGTM.

andela-abdulsemiu committed:
pageCtrl update

Notifies user when an edge case is detected

abdulsemiu-atanda committed:
README.md Update

andela-abdulsemiu committed:
Edge case update in Create Index method

andela-abdulsemiu committed:
Merge branch 'development' of https://github.com/andela-aatanda/checkpoint1-inverted-index into development

andela-abdulsemiu committed:
Test Suite Update

andela-abdulsemiu committed:
browserify test file ignore

andela-abdulsemiu committed:
browserify task update

andela-abdulsemiu committed:
browserify test file update

andela-abdulsemiu committed:
JSON file update

andela-abdulsemiu committed:
Dev Dependencies update

andela-abdulsemiu committed:
ES6 syntax update

andela-abdulsemiu committed:
Test spec update

andela-abdulsemiu committed:
Test Update

Dependency update in package.json
test task in gulpfile.js
karma configuration update

andela-abdulsemiu committed:
task update

andela-abdulsemiu committed:
missing semi-colon fix

andela-abdulsemiu committed:
Verbose reporter fix

Dependency update in package.json
karma config update

andela-abdulsemiu committed:
ES6 conformity configuration update

andela-abdulsemiu committed:
View update to make create index conspicous
